packages/runtime/src/runtime/entry.tsx

```diff
@@ -20,24 +21,35 @@ export const EnsembleEntry: React.FC<EnsembleEntryProps> = ({
 
   const hasMenu = "items" in entry;
   useEffect(() => {
-    if (screen && location.pathname !== `/${screen.name.toLowerCase()}`) {
-      navigate({
-        pathname: `/${screen.name.toLowerCase()}`,
-        search: location.search,
-      });
+    if (!keys(entry).length) {
       return;
     }
 
-    if (!(hasMenu && location.pathname === "/")) {
+    const navigateToPath = (pathname: string, search = ""): void => {
+      navigate({ pathname, search });
+    };
+
+    const getLowerCasePath = (name: string): string => `/${name.toLowerCase()}`;
+
+    const initialScreen = location.pathname;
+
+    if (initialScreen !== "/") {
+      return;
+    }
+
```

Comment:
This is incorrect. We should navigate to the path specified in the URL

Reply:
You mean `location.pathname` ?

Reply:
Can you leave a comment why this works?